parquet-column/src/main/java/org/apache/parquet/column/values/factory/ValuesWriterFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.values.factory;
+
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.column.values.ValuesWriter;
+
+/**
+ * Can be overridden to allow users to specify how they want their ValuesWriters to be created.
+ * ValuesWriterFactories are created using reflection in {@link org.apache.parquet.column.ParquetProperties}.
+ * ValuesWriterFactories can in turn read additional config to create appropriate ValuesWriters.
+ */
+public interface ValuesWriterFactory {
+
+  /**
+   * Used to initialize the factory. This method is called before newValuesWriter()
+   */
+  void initialize(ValuesWriterFactoryParams params);
+
+  /**
+   * Creates a ValuesWriter to help write the given column.
+   */
+  ValuesWriter newValuesWriter(ColumnDescriptor descriptor);
```

Comment:
So as of now we will have:
1) ValuesWriterParams - things like writerVer, initialCapacity, pageSize, enableDictionary, maxDictionarySize
2) Hadoop conf (if needed)
3) ColumnDescriptor - this has column path, type, maxRep, maxDef. 

Guess the only other thing I can think of is the actual data in columns but we can redirect that to the actual values writer (basically what I was getting at above with the fallback based approach).